controllers/mysql_container.go

```diff
@@ -6,287 +6,275 @@ import (
-			},
-			{
-				MountPath: constants.LogDirPath,
-				Name:      constants.VarLogVolumeName,
-			},
-		},
-	}
-	updateContainerWithSupplements(&c, sts.Spec.Template.Spec.Containers)
+	c := corev1ac.Container().
+		WithName(constants.SlowQueryLogAgentContainerName).
+		WithImage(r.FluentBitImage).
+		WithVolumeMounts(
+			corev1ac.VolumeMount().
+				WithName(constants.SlowQueryLogAgentConfigVolumeName).
+				WithMountPath(constants.FluentBitConfigPath).
+				WithReadOnly(true),
+			corev1ac.VolumeMount().
+				WithName(constants.VarLogVolumeName).
+				WithMountPath(constants.LogDirPath),
+		)
+
+	updateContainerWithSecurityContext(c)
+
 	return c
 }
 
-func (r *MySQLClusterReconciler) makeV1ExporterContainer(collectors []string, current []corev1.Container) corev1.Container {
-	c := corev1.Container{
-		Name:  constants.ExporterContainerName,
-		Image: r.ExporterImage,
-		Ports: []corev1.ContainerPort{
-			{ContainerPort: constants.ExporterPort, Name: constants.ExporterPortName, Protocol: corev1.ProtocolTCP},
-		},
-		VolumeMounts: []corev1.VolumeMount{
-			{
-				MountPath: constants.RunPath,
-				Name:      constants.RunVolumeName,
-			},
-			{
-				MountPath: constants.MyCnfSecretPath,
-				Name:      constants.MySQLConfSecretVolumeName,
-				ReadOnly:  true,
-			},
-		},
-	}
-	c.Args = []string{"--config.my-cnf=" + filepath.Join(constants.MyCnfSecretPath, constants.ExporterMyCnf)}
+func (r *MySQLClusterReconciler) makeV1ExporterContainer(collectors []string) *corev1ac.ContainerApplyConfiguration {
+	c := corev1ac.Container().
+		WithName(constants.ExporterContainerName).
+		WithImage(r.ExporterImage).
+		WithArgs("--config.my-cnf="+filepath.Join(constants.MyCnfSecretPath, constants.ExporterMyCnf)).
+		WithPorts(
+			corev1ac.ContainerPort().
+				WithName(constants.ExporterPortName).
+				WithContainerPort(constants.ExporterPort).
+				WithProtocol(corev1.ProtocolTCP)).
+		WithVolumeMounts(
+			corev1ac.VolumeMount().
+				WithName(constants.RunVolumeName).
+				WithMountPath(constants.RunPath),
+			corev1ac.VolumeMount().
+				WithName(constants.MySQLConfSecretVolumeName).
+				WithMountPath(constants.MyCnfSecretPath).
+				WithReadOnly(true),
+		)
+
 	for _, cl := range collectors {
-		c.Args = append(c.Args, "--collect."+cl)
+		c.WithArgs("--collect." + cl)
 	}
 
-	updateContainerWithSupplements(&c, current)
+	updateContainerWithSecurityContext(c)
+
 	return c
 }
 
-func (r *MySQLClusterReconciler) makeV1OptionalContainers(cluster *mocov1beta2.MySQLCluster, current []corev1.Container) []corev1.Container {
-	var containers []corev1.Container
-	for _, c := range cluster.Spec.PodTemplate.Spec.Containers {
-		switch c.Name {
+func (r *MySQLClusterReconciler) makeV1OptionalContainers(cluster *mocov1beta2.MySQLCluster) []*corev1ac.ContainerApplyConfiguration {
+	var containers []*corev1ac.ContainerApplyConfiguration
+
+	spec := cluster.Spec.PodTemplate.Spec.DeepCopy()
+	for _, c := range spec.Containers {
+		c := c
+		switch *c.Name {
```

Comment:
Please add a nil check. The moco-controller will be crashed by the following manifest. 

```yaml
apiVersion: moco.cybozu.com/v1beta2
kind: MySQLCluster
metadata:
  name: foo
spec:
  podTemplate:
    spec:
      containers:
      - image: quay.io/cybozu/mysql:8.0.27
        name: mysqld
      - image: invalid      # nameがなく不正なコンテナ定義
  volumeClaimTemplates:
  - metadata:
      name: mysql-data
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 1Gi
  logRotationSchedule: "@every 10s"
```